css-viewport/Overview.bs

```diff
@@ -356,6 +360,157 @@ only the value previously set to it.
 }
 </pre>
 
+<h2 id='zoom-property'>
+	The 'zoom' property
+</h2>
+
+An element becomes zoomed when the 'zoom' property has a positive computed value different than 1 (or when a flat tree ancestor has zoom).
+'zoom' affects computed property values and inherited property values.
```

Comment:
The zoom/effective-zoom doesn't affect any computed property values?

If I understand the expected effects correctly, it's a lot easier for me to think of it as an effect on the CSS reference pixel on an element basis. Then it doesn't really change cascading/ inheritance?

Reply:
> The zoom/effective-zoom doesn't affect any computed property values?

It does. It just doesn't affect the serialized form returned by getComputedStyle().

> If I understand the expected effects correctly, it's a lot easier for me to think of it as an effect on the CSS reference pixel on an element basis.

You could think of it that way...

> Then it doesn't really change cascading/ inheritance?

Defining it that way might simplify the definition somewhat, but it's already only a couple of sentences. Is there something very unclear about how I've written it?

Reply:
> > The zoom/effective-zoom doesn't affect any computed property values?
> 
> It does. It just doesn't affect the serialized form returned by getComputedStyle().

I don't think so. Then it would affect the output from typed OM, and I don't think we want that.

In Chrome's implementation the effective zoom factor is applied to px values on the ComputedStyle objects, and it also includes the device to css pixel ratio baked in. That's an implementation detail. It doesn't mean computed values are affected by the device pixel ratio.

> > If I understand the expected effects correctly, it's a lot easier for me to think of it as an effect on the CSS reference pixel on an element basis.
> 
> You could think of it that way...
> 
> > Then it doesn't really change cascading/ inheritance?
> 
> Defining it that way might simplify the definition somewhat, but it's already only a couple of sentences. Is there something very unclear about how I've written it?

I find the =scale=/ =unscale= concept a bit hard to verify.

Reply:
> I don't think so. Then it would affect the output from typed OM, and I don't think we want that.
> 
> In Chrome's implementation the effective zoom factor is applied to px values on the ComputedStyle objects, and it also includes the device to css pixel ratio baked in. That's an implementation detail. It doesn't mean computed values are affected by the device pixel ratio.

Yes...maybe I am not understanding what you are asking. Maybe I should say "used style" instead of "computed style"?

Reply:
Used values are what the resolved values refer to for e.g. the width property, so that wouldn't work directly without also changing the definition of resolved values in CSSOM. I'm not even sure actual values include the device pixel ratio. CSS has a concept of device pixels used e.g. in https://drafts.csswg.org/css-values-4/#ref-for-device-pixel

Maybe reference pixel isn't the right term here, but need to use the CSS px to device pixels somehow.

Reply:
Ok I read through the specs. I have concluded that it should apply to the used value but just before layout; spec text adjusted accordingly. Why:

* The CSS reference pixel is for defining the ratio between a generic CSS pixel (before any zoom is applied) and device pixels. This concept is really a post-layout one, and also is a global concept and not specific to an element. For these reasons, we can't amend that.

* As you said, modifying computed styles would affect CSSOM APIs by default, so we don't want that if possible.

* The actual size is post-layout, which we can't use for that reason, because zoom affects layout.

* The used size is where units like `auto` get resolved to numbers via layout. I placed zoom at the start of the used size calculation. We could in principle define yet another stage between computed and used just for zoom, but that seems like overkill.